packages/console/src/hooks/use-current-tenant-scopes.ts

```diff
@@ -1,61 +1,56 @@
+import useCurrentUser from './use-current-user';
+
 const useCurrentTenantScopes = () => {
   const { currentTenantId, isInitComplete } = useContext(TenantsContext);
-  const { isAuthenticated, getOrganizationTokenClaims } = useLogto();
-
-  const [scopes, setScopes] = useState<string[]>([]);
-  const [canInviteMember, setCanInviteMember] = useState(false);
-  const [canRemoveMember, setCanRemoveMember] = useState(false);
-  const [canUpdateMemberRole, setCanUpdateMemberRole] = useState(false);
-  const [canManageTenant, setCanManageTenant] = useState(false);
-
-  useEffect(() => {
-    (async () => {
-      if (isAuthenticated && isInitComplete) {
-        const organizationId = getTenantOrganizationId(currentTenantId);
-        const claims = await getOrganizationTokenClaims(organizationId);
-        const allScopes = claims?.scope?.split(' ') ?? [];
-        setScopes(allScopes);
-
-        for (const scope of allScopes) {
-          switch (scope) {
-            case TenantScope.InviteMember: {
-              setCanInviteMember(true);
-              break;
-            }
-            case TenantScope.RemoveMember: {
-              setCanRemoveMember(true);
-              break;
-            }
-            case TenantScope.UpdateMemberRole: {
-              setCanUpdateMemberRole(true);
-              break;
-            }
-            case TenantScope.ManageTenant: {
-              setCanManageTenant(true);
-              break;
-            }
-            default: {
-              break;
-            }
-          }
-        }
-      }
-    })();
-  }, [currentTenantId, getOrganizationTokenClaims, isAuthenticated, isInitComplete]);
-
-  return {
-    canInviteMember,
-    canRemoveMember,
-    canUpdateMemberRole,
-    canManageTenant,
-    scopes,
-  };
+  const cloudApi = useAuthedCloudApi();
+  const { user } = useCurrentUser();
+  const userId = user?.id ?? '';
+
+  const { data, isLoading, mutate } = useSWR<OrganizationScope[], RequestError>(
+    userId && isInitComplete && `api/tenants/${currentTenantId}/members/${userId}/scopes`,
+    async () =>
+      cloudApi.get('/api/tenants/:tenantId/members/:userId/scopes', {
+        params: { tenantId: currentTenantId, userId },
+      })
+  );
+
+  const scopes = useMemo(() => data?.map(({ name }) => name), [data]);
+
+  const userTenantAccesses = useMemo(() => {
+    if (!scopes || scopes.length === 0) {
+      return {
+        canInviteMember: false,
+        canRemoveMember: false,
+        canUpdateMemberRole: false,
+        canManageTenant: false,
+      };
+    }
+
+    return {
+      canInviteMember: scopes.includes(TenantScope.InviteMember),
+      canRemoveMember: scopes.includes(TenantScope.RemoveMember),
+      canUpdateMemberRole: scopes.includes(TenantScope.UpdateMemberRole),
+      canManageTenant: scopes.includes(TenantScope.ManageTenant),
+    };
+  }, [scopes]);
+
+  return useMemo(
+    () => ({
+      isLoading,
+      scopes,
+      ...userTenantAccesses,
```

Comment:
a separate key will be more clear

```suggestion
      access,
```

Reply:
ok